pkg/common/consumer/consumer_handler.go

```diff
@@ -73,6 +73,15 @@ func (consumer *SaramaConsumerHandler) ConsumeClaim(session sarama.ConsumerGroup
 	// The `ConsumeClaim` itself is called within a goroutine, see:
 	// https://github.com/Shopify/sarama/blob/master/consumer_group.go#L27-L29
 	for message := range claim.Messages() {
+
+		// Exit if the session has been cancelled due to either rebalancing
+		// or adapter being shutting down
+		select {
+		case <-session.Context().Done():
```

Comment:
You don't need to add this code, you should invoke `consumerGroup.Close()`. sarama closes the`claim.Messages()` when `consumerGroup.Close()` is invoked

Reply:
It's up to the `consumerGroup` owner to listen on a context and invoke `consumerGroup.Close()` when a specific context is finished

Reply:
ok cool thanks! I'll look into this.

Reply:
I removed it.